### PR TITLE
RUM-9631: Add accessibility attributes

### PR DESCRIPTION
**Pull request by jonathanmos**
### What does this PR do?
Sets up a feature flag (`collectAccessibilitySettings`). When this flag is active, the SDK collects a number of custom accessibility attributes to send in view updates under the "accessibility" tag. These attributes are:

closed_captions_enabled (boolean) - has closed captioning been activated
color_inversion_enabled (boolean) - has color inversion been activated
reduced_animations_enabled (boolean) - has reduced animation been activated
screen_reader_enabled (boolean) - has talkback been activated
single_app_mode_enabled (boolean) - has screen pinning been activated
text_size (float) - current font size (1.0 is "normal" font)

Some attribute values are collected with listeners that record the new value whenever it changes. For attributes where this is not possible because no listener exists, the value is polled - but not more than once every 30 seconds.

### Motivation
Collect additional user insights.

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt [hunk 1]
```diff
@@ -285,6 +285,14 @@ data class RumConfiguration internal constructor(
             return this
         }
 
+        /**
+         * Sets a flag to collect accessibility settings inside the RUM view end event.
+         */
+        fun collectAccessibilitySettings(): Builder {
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt]: Just went through the similar APIs in the `RumConfiguration`,  we have: 
```
fun trackNonFatalAnrs(enabled: Boolean)
fun trackBackgroundEvents(enabled: Boolean)
fun trackFrustrations(enabled: Boolean)
fun trackAnonymousUser(enabled: Boolean)
```

In this case should we also add `enabled` argument in this function?

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt [hunk 2]
```diff
@@ -285,6 +285,14 @@ data class RumConfiguration internal constructor(
             return this
         }
 
+        /**
+         * Sets a flag to collect accessibility settings inside the RUM view end event.
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt]: Better to explain default behavior which is disabled.

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/Accessibility.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        textSize?.let { put(TEXT_SIZE_KEY, it) }
+        isScreenReaderEnabled?.let { put(SCREEN_READER_ENABLED_KEY, it) }
+        isColorInversionEnabled?.let { put(COLOR_INVERSION_ENABLED_KEY, it) }
+        isClosedCaptioningEnabled?.let { put(CLOSED_CAPTIONING_ENABLED_KEY, it) }
+        isReducedAnimationsEnabled?.let { put(REDUCED_ANIMATIONS_ENABLED_KEY, it) }
+        isScreenPinningEnabled?.let { put(SCREEN_PINNING_ENABLED_KEY, it) }
+    }
+
+    companion object {
+        internal val EMPTY_STATE = Accessibility(
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/Accessibility.kt]: It can just be `Accessibility()` here ?

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/Accessibility.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            isReducedAnimationsEnabled = null,
+            isScreenPinningEnabled = null
+        )
+
+        internal const val TEXT_SIZE_KEY = "text_size"
+        internal const val SCREEN_READER_ENABLED_KEY = "screen_reader_enabled"
+        internal const val COLOR_INVERSION_ENABLED_KEY = "invert_colors_enabled"
+        internal const val CLOSED_CAPTIONING_ENABLED_KEY = "closed_captioning_enabled"
+        internal const val REDUCED_ANIMATIONS_ENABLED_KEY = "reduced_animations_enabled"
+        internal const val SCREEN_PINNING_ENABLED_KEY = "single_app_mode_enabled"
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/Accessibility.kt:42]: why the value of `SCREEN_PINNING_ENABLED_KEY` is "single_app_mode_enabled"?

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt [hunk 5]
```diff
@@ -950,7 +955,16 @@ internal open class RumViewScope(
         val isSlowRendered = resolveRefreshRateInfo(refreshRateInfo) ?: false
         // make a copy - by the time we iterate over it on another thread, it may already be changed
         val eventFeatureFlags = featureFlags.toMutableMap()
+
         val eventAdditionalAttributes = (eventAttributes + globalAttributes).toMutableMap()
+
+        accessibilityReader.let { reader ->
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt]: `let` seems not necessary here

File: features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt [hunk 6]
```diff
@@ -0,0 +1,956 @@
...
+    fun `M maintain current state W getState { polling does not change values }`() {
+        // Given
+        val fontScale = 1.2f
+        mockConfiguration.fontScale = fontScale
+        whenever(mockAccessibilityManager.isTouchExplorationEnabled) doReturn true
+        whenever(mockActivityManager.lockTaskModeState) doReturn ActivityManager.LOCK_TASK_MODE_NONE
+
+        // When
+        val firstResult = testedReader.getState()
+        val secondResult = testedReader.getState()
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt]: is this a false positive? the getState value doesn't change not only because the value is not changed but also the threshold is not reached?

File: features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt [hunk 7]
```diff
@@ -0,0 +1,956 @@
...
+
+    // region Edge Cases
+    @Test
+    fun `M return partial state W getState { some managers are null }`() {
+        // Given
+        testedReader = DatadogAccessibilityReader(
+            internalLogger = mockInternalLogger,
+            applicationContext = mockContext,
+            resources = null,
+            activityManager = null,
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt]: do you know in reality what's the reason if  `activityManager` or `accessibilityManager` is null?

File: features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt [hunk 8]
```diff
@@ -0,0 +1,956 @@
...
+        // Given
+        val executor = java.util.concurrent.Executors.newFixedThreadPool(3)
+        val results = java.util.concurrent.ConcurrentLinkedQueue<Map<String, Any>>()
+        val countDownLatch = java.util.concurrent.CountDownLatch(3)
+
+        // When - Multiple threads call getState concurrently
+        repeat(3) {
+            executor.submit {
+                try {
+                    val result = testedReader.getState()
```
Comment (ambushwork) [inline features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt]: If I am not mistake, `getState()` returns always the same result here?

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/Accessibility.kt (hunk 4 above)
Comment (jonathanmos) [reply features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/Accessibility.kt:42]: The values are for parity with what ios is sending

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt (hunk 1 above)
Comment (jonathanmos) [reply features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt]: Following discussion, I'm going to make this internal and move it into the _RumInternalProxy to make this less customer facing and simple to remove later

File: features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt (hunk 7 above)
Comment (jonathanmos) [reply features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReaderTest.kt]: I've removed these tests because they weren't helpful

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt (hunk 1 above)
Comment (ambushwork) [reply features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt]: Why we need to make this internal and remove later? is it because that this is an experimental API?

Comment (jonathanmos) [reply features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt]: This api is only to allow us to gather initial data. When the feature is rolled out it would be active for everyone

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReader.kt [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+import android.view.accessibility.AccessibilityManager.TouchExplorationStateChangeListener
+import com.datadog.android.api.InternalLogger
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicLong
+
+@Suppress("TooManyFunctions")
+internal class DatadogAccessibilityReader(
+    private val internalLogger: InternalLogger,
+    private val applicationContext: Context,
+    private val resources: Resources? = Resources.getSystem(),
```
Comment (aleksandr-gringauz) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReader.kt]: Why do we need to use `Resources.getSystem` instead of `applicationContext.resources` here?

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt (hunk 1 above)
Comment (ambushwork) [reply features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt]: Then shouldn't we use `@ExperimentalRumApi` instead?

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReader.kt (hunk 9 above)
Comment (jonathanmos) [reply features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/accessibility/DatadogAccessibilityReader.kt]: I think you are right - good catch. I'll fix this

**Review by ambushwork — APPROVED** [review]

**Review by aleksandr-gringauz — APPROVED** [review]
